src/colorizer/Dataset.ts

```diff
@@ -9,17 +9,41 @@ import Track from "./Track";
 
 import { FeatureArrayType, FeatureDataType } from "./types";
 import * as urlUtils from "./utils/url_utils";
+import { MAX_FEATURE_CATEGORIES } from "../constants";
+
+export enum FeatureType {
+  CONTINUOUS = "continuous",
+  DISCRETE = "discrete",
+  CATEGORICAL = "categorical",
+}
 
 export type FeatureData = {
   data: Float32Array;
   tex: Texture;
   min: number;
   max: number;
   units: string;
+  type: FeatureType;
+  categories: string[] | null;
 };
 
-export type FeatureMetadata = {
-  units: string | null;
+/**
+ * JSON metadata for dataset features
+ * This is the deprecated version, where feature metadata
+ * was stored separately from the feature file path declaration.
+ */
+type DeprecatedFeatureMetadata = {
+  units?: string | null;
+  type?: string | null;
+  categories?: string[] | null;
+};
+
+/** JSON metadata for dataset features. */
+type FeatureMetadata = {
```

Comment:
Nit: regarding naming, it's slightly uncomfortable that `FeatureData` is the runtime datatype that the app uses but `FeatureMetadata` is the json file format datatype.  I'm not sure if you can namespace things or decorate the name with "FileFormat" or something.